src/ds5/ds5-options.cpp

```diff
@@ -37,6 +37,25 @@ namespace librealsense
                         "Emitter select, 0-disable all emitters, 1-enable laser, 2-enable laser auto (opt), 3-enable LED (opt)")
     {}
 
+    emitter_frequency::emitter_frequency(
+        uvc_sensor & ep, const std::map< float, std::string > & description_per_value )
+        : uvc_xu_option( ep,
+                         ds::depth_xu,
+                         ds::DS5_EMITTER_FREQUENCY,
+                         "Controls the emitter frequency, 57 [KHZ] / 91 [KHZ]",
+                         description_per_value )
+    {
+    }
+
+    void emitter_frequency::set( float value )
+    {
+        if( _ep.is_streaming() )
+            throw wrong_api_call_sequence_exception( "Setting the emitter frequency during streaming is not allowed" );
+
+        uvc_xu_option::set( value );
```

Comment:
recording function seems to be used in the other options' set method - It may need to be called here too
_recording_function(*this);

Reply:
It does nothing, so I didn't add it
`std::function<void(const option&)> _recording_function = [](const option&) {};`

Reply:
seems that this is the default value.
If you call the enable_recording method, you can set what will be the recording action:
void enable_recording(std::function<void(const option&)> record_action) override
{
        _recording_function = record_action;
}